service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformManager.java

```diff
@@ -89,8 +88,10 @@ private static String executeTerraformCommand(Command command, Map<String, Strin
         switch (commandVerb) {
             case "apply" -> {
                 writeHclToFile(command, configPath);
-                terraformClient.plan(envVars).get();
-                terraformClient.apply(envVars).get();
+                Boolean planSuccessful = terraformClient.plan(envVars).get();
```

Comment:
No point in running apply if the plan failed. This short circuit saves us 255 seconds on a config push request.